### ISSUE TITLE
Add `declaration-block-trailing-semicolon` support for nested properties and shorthand values.

### ISSUE THREAD

**Issue by pamelalozano16** [stylelint-scss/stylelint-scss#1090]
`declaration-block-trailing-semicolon` support for nested properties / shorthand values.

This example: 
```
div {
  border: solid #000 {
    width: 1px 1px 1px 7px;
  }
}
```
Triggers a false positive: `line 2, col 20 - Expected a trailing semicolon (declaration-block-trailing-semicolon)`

**Comment by kristerkari** [comment]
I'm not sure how I feel about this one since Stylelint has deprecated all rules that have to do with styling. 

Before Prettier it was quite common to have stylistic linting rules, but since Prettier became popular, a lot of the stylistic stuff has been deprecated and moved into `stylelint-stylistic`. 

There has been discussion about it in this repo too (#814), but I have personally been to lazy to do the work to move the rules into a separate package.

Maybe it would be time to finally do that, and then you can open a PR against that package.

**Comment by pamelalozano16** [comment]
Unfortunately, I don’t have the bandwidth to migrate all the rules to `stylelint-stylistic`, but I’m happy to move this PR once that package is available.

**Comment by kristerkari** [comment]
Don't worry, I can take care of the migration when I have a free moment. I think that it would be beneficial to do the sepration in the same way that Stylelint did already a long time ago.
